src/main/webapp/app/course/competencies/competency-management/competency-management-table.component.ts

```diff
@@ -0,0 +1,135 @@
+import { faFileImport, faPencilAlt, faPlus, faTrash } from '@fortawesome/free-solid-svg-icons';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ImportAllCompetenciesComponent, ImportAllFromCourseResult } from 'app/course/competencies/competency-management/import-all-competencies.component';
+import { PrerequisiteService } from 'app/course/competencies/prerequisite.service';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisMarkdownModule } from 'app/shared/markdown.module';
+
+@Component({
+    selector: 'jhi-competency-management-table',
+    templateUrl: './competency-management-table.component.html',
+    standalone: true,
+    imports: [ArtemisSharedModule, ArtemisMarkdownModule],
+})
+export class CompetencyManagementTableComponent implements OnInit, OnDestroy {
+    @Input() courseId: number;
+    @Input() courseCompetencies: CourseCompetency[];
+    @Input() relations: CompetencyRelation[];
+    @Input() competencyType: CourseCompetencyType;
+    @Input() standardizedCompetenciesEnabled: boolean;
+
+    @Output() competencyDeleted = new EventEmitter<number>();
+
+    service: CompetencyService | PrerequisiteService;
+    private dialogErrorSource = new Subject<string>();
+    dialogError = this.dialogErrorSource.asObservable();
+    pluralCompetencyType: 'competencies' | 'prerequisites';
+
+    // Injected services
+    private readonly competencyService: CompetencyService = inject(CompetencyService);
+    private readonly prerequisiteService: PrerequisiteService = inject(PrerequisiteService);
+    private readonly alertService: AlertService = inject(AlertService);
+    private readonly modalService: NgbModal = inject(NgbModal);
+
+    readonly faFileImport = faFileImport;
+    readonly faPlus = faPlus;
+    readonly faPencilAlt = faPencilAlt;
+    readonly faTrash = faTrash;
+
+    readonly getIcon = getIcon;
+
+    ngOnInit(): void {
+        if (this.competencyType === CourseCompetencyType.COMPETENCY) {
+            this.service = this.competencyService;
+            this.pluralCompetencyType = 'competencies';
+        } else {
+            this.service = this.prerequisiteService;
+            this.pluralCompetencyType = 'prerequisites';
+        }
+    }
+
+    ngOnDestroy() {
+        this.dialogErrorSource.unsubscribe();
+    }
+
+    /**
+     * Opens a modal for selecting a course to import all competencies from.
+     */
+    openImportAllModal() {
+        const modalRef = this.modalService.open(ImportAllCompetenciesComponent, { size: 'lg', backdrop: 'static' });
+        //unary operator is necessary as otherwise courseId is seen as a string and will not match.
+        modalRef.componentInstance.disabledIds = [+this.courseId];
+        modalRef.componentInstance.competencyType = this.competencyType;
+        modalRef.result.then((result: ImportAllFromCourseResult) => {
+            const courseTitle = result.courseForImportDTO.title ?? '';
+
+            this.service
```

Comment:
can we maybe add this to a subscription and unsubscribe in ngOnDestroy?

Reply:
As far as I can see, we usually don't do it for other rest calls as well. Is there a good reason, why we should start now?

Reply:
yes your right we rather focus on other reasons to unsubscribe 👍